fingertip/plugins/os/centos-stream.py

```diff
@@ -0,0 +1,104 @@
+import os
+
+import fingertip.machine
+from fingertip.util import path
+
+#[Repos]
+#http://ftp.fi.muni.cz/pub/linux/centos/8-stream/
+#official mirror: http://mirror.centos.org/centos/8-stream/
+
+REPO = 'http://mirror.centos.org/centos/'
+
+def main(m=None, fips=False):
+    m = m or fingertip.build('backend.qemu')
+    if hasattr(m, 'qemu'):
+        return m.apply(install_in_qemu, fips=fips)
+    raise NotImplementedError()
+
+
+def install_in_qemu(m=None,
+                    fips=False):
+    """
+    :param bool fips: if True, install system in FIPS mode
+    """
+
+    m = m or fingertip.machine.build('backend.qemu')
+    URL = 'http://mirror.centos.org/centos/8-stream/BaseOS/x86_64/os/'
+
+    with m:
+        ssh_key_path = path.fingertip('ssh_key', 'fingertip.pub')
+        with open(ssh_key_path) as f:
+            ssh_pubkey = f.read().strip()
+        m.expiration.depend_on_a_file(ssh_key_path)
+
+        ks = path.fingertip('fingertip', 'plugins', 'os', 'centos-stream.ks')
+        HOSTNAME = 'centos'
+
+        if fips:
+            HOSTNAME += '-fips'
+
+        fqdn = HOSTNAME + '.fingertip.local'
+
+        with open(ks) as f:
+            ks_text = f.read().format(HOSTNAME=fqdn,
+                                      SSH_PUBKEY=ssh_pubkey,
+                                      PROXY=m.http_cache.internal_url)
+        m.expiration.depend_on_a_file(ks)
+        m.http_cache.mock('http://mock/ks', text=ks_text)
+        m.log.info(f'fetching kernel: {URL}/isolinux/vmlinuz')
+        kernel = os.path.join(m.path, 'kernel')
+        m.http_cache.fetch(f'{URL}/isolinux/vmlinuz', kernel)
+        m.log.info(f'fetching initrd: {URL}/isolinux/initrd.img')
+        initrd = os.path.join(m.path, 'initrd')
+        m.http_cache.fetch(f'{URL}/isolinux/initrd.img', initrd)
+        append = ('ks=http://mock/ks console=ttyS0 ' + 'inst.text ' + 'inst.notmux ' + 'inst.cmdline ' +
```

Comment:
1) keep lines shorter than 79 characters
2) why do that `'a ' + 'b ' + 'c '` when you can do `'a b c '`?